lib/AbstractPlugin.js

```diff
@@ -2,21 +2,26 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function AbstractPlugin(plugins) {
-	this._plugins = plugins || {};
-}
-module.exports = AbstractPlugin;
+"use strict";
+
+class AbstractPlugin {
+	static create(plugins) {
+		return class Plugin extends AbstractPlugin {
+			constructor() {
+				super(plugins);
+			}
+		};
+	}
 
-AbstractPlugin.create = function(plugins) {
-	function Plugin() {
-		AbstractPlugin.call(this, plugins);
+	constructor(plugins) {
+		this._plugins = plugins || {};
```

Comment:
It'd be nice to convert this into a `Map`, but that'd be a larger scale refactoring